core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -425,35 +425,63 @@ class KafkaApis(val requestChannel: RequestChannel,
       requestHelper.sendMaybeThrottle(request, offsetCommitRequest.getErrorResponse(Errors.UNSUPPORTED_VERSION.exception))
       CompletableFuture.completedFuture[Unit](())
     } else {
+      val topicNames =
+        if (offsetCommitRequest.version() >= 9)
+          metadataCache.topicIdsToNames()
+        else
+          Collections.emptyMap[Uuid, String]()
+
+      val resolvedTopics = new ArrayBuffer[OffsetCommitRequestData.OffsetCommitRequestTopic]()
+      offsetCommitRequest.data.topics.forEach { topic =>
+        var topicName = topic.name()
+        if (Utils.isBlank(topicName)) {
+          // Expected for requests version >= 9 which rely on topic IDs exclusively.
+          topicName = topicNames.get(topic.topicId())
+        }
+        if (topicName != null) {
+          topic.setName(topicName)
+          resolvedTopics += topic
+        }
+      }
+
       val authorizedTopics = authHelper.filterByAuthorized(
         request.context,
         READ,
         TOPIC,
-        offsetCommitRequest.data.topics.asScala
+        resolvedTopics
       )(_.name)
 
-      val responseBuilder = new OffsetCommitResponse.Builder()
+      val responseBuilder = OffsetCommitResponse.newBuilder(offsetCommitRequest.version())
       val authorizedTopicsRequest = new mutable.ArrayBuffer[OffsetCommitRequestData.OffsetCommitRequestTopic]()
       offsetCommitRequest.data.topics.forEach { topic =>
```

Comment:
You could use `resolvedTopics` instead of `offsetCommitRequest.data.topics` here.